init-tools.sh

```diff
@@ -132,7 +132,13 @@ if [ ! -e $__DOTNET_PATH ]; then
         cd $__DOTNET_PATH
         tar -xf $__DOTNET_PATH/dotnet.tar
     }
-    execute_with_retry install_dotnet_cli >> "$__init_tools_log" 2>&1
+
+    if [[ "$DotNetBootstrapCliTarPath" = "" ]]; then
```

Comment:
Wouldn't it make since to do this inside install_dotnet_cli? that way we get retry logic if the un-taring fails for some reason?

Reply:
If we don't need the retry it might be better to skip all the other stuff under `if [ ! -e $__DOTNET_PATH ];` and just the extraction above it. Assuming none of the other variables are needed.

Reply:
Retry is useful for network outages.  I was assuming tar would be on local file system, so it was unnecessary.  The retry should cause any harm though...

When I was originally testing on x64, the __PKG_* variables were needed to initialize buildtools below.  That was why the original attempt was so messy.  With the ILAsm disable below, I am not sure if they are still needed, but the core is simplest if this is kept close to the `curl`/`wget`

Reply:
I think it will move nicely into `install_dotnet_cli`